### PR TITLE
RSM-014 schema validation test

### PR DESCRIPTION
**Pull request by lasrinnil**
## References

* #0000

### CODE REVIEW

File: source/dotnet/Tests/Sender/Infrastructure/Services/DocumentFactoryTests.cs [hunk 1]
```diff
@@ -80,6 +82,89 @@ public class DocumentFactoryTests
...
+            documentIdGeneratorMock,
+            seriesIdGeneratorMock,
+            processRepositoryMock,
+            storageHandlerMock);
+
+        const string expectedIsoString = "2022-07-04T08:05:30Z";
+        var instant = InstantPattern.General.Parse(expectedIsoString).Value;
+        clockMock
+            .Setup(clock => clock.GetCurrentInstant())
+            .Returns(instant);
```
Comment (BjarkeMeier) [inline source/dotnet/Tests/Sender/Infrastructure/Services/DocumentFactoryTests.cs:146]: How about moving this to `MockServices` as well?

File: source/dotnet/Tests/Sender/Infrastructure/Services/DocumentFactoryTests.cs [hunk 2]
```diff
@@ -80,6 +82,89 @@ public class DocumentFactoryTests
...
+
+        await using var outStream = new MemoryStream();
+
+        // Act
+        await sut.CreateAsync(request, outStream);
+        outStream.Position = 0;
+
+        // Assert
+        var reader = new SchemaValidatingReader(outStream, Schemas.CimXml.MeasureNotifyAggregatedMeasureData);
+        Assert.False(reader.HasErrors);
```
Comment (BjarkeMeier) [inline source/dotnet/Tests/Sender/Infrastructure/Services/DocumentFactoryTests.cs:156]: nit:
```suggestion
        reader.HasErrors.Should().BeFalse();
```

**Review by BjarkeMeier — APPROVED** [review]

File: source/dotnet/Tests/Sender/Infrastructure/Services/DocumentFactoryTests.cs (hunk 1 above)
Comment (lasrinnil) [reply source/dotnet/Tests/Sender/Infrastructure/Services/DocumentFactoryTests.cs:146]: Oh yeah, the expected string was where I drew the line :D

Comment (BjarkeMeier) [reply source/dotnet/Tests/Sender/Infrastructure/Services/DocumentFactoryTests.cs:146]: How about still moving but making expected an argument?

Comment (BjarkeMeier) [reply source/dotnet/Tests/Sender/Infrastructure/Services/DocumentFactoryTests.cs:146]: You decide

**Pull request merged**